multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -416,9 +407,15 @@ func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
 	defer klog.InfoS("Shutting down StaleResCleanupController")
 
 	ctx, _ := wait.ContextForChannel(stopCh)
-	if err := c.RunOnce(ctx); err != nil {
-		c.Enqueue()
-		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+
+	switch c.clusterRole {
+	case LeaderCluster:
+		go wait.UntilWithContext(ctx, c.cleanupMemberClusterAnnounces, memberClusterAnnounceStaleTime/2)
+	case MemberCluster:
+		if err := c.RunOnce(ctx); err != nil {
+			c.Enqueue()
+			go wait.UntilWithContext(ctx, c.runWorker, 5*time.Second)
```

Comment:
We should stop running runWorker() when all stale resources are successfully cleaned up?

Reply:
Done.